pkgs/applications/networking/modem-manager-gui/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv, buildEnv, pkgconfig, python3, fetchhg, gtk3-x11, glib, gdbm, gtkspell3, itstool, libappindicator-gtk3, meson, ninja, perldevelPackages }:
+
+stdenv.mkDerivation rec {
+  name = "modem-manager-gui";
+  version = "0.4.17";
+  rev = "v${version}";
+
+  src = fetchhg {
+    url = https://linuxonly@bitbucket.org/linuxonly/modem-manager-gui;
+    rev = "8bcf842";
+    sha256 = "04p08gzvi068bd3slbnaghajwzp4h5jacib640k5shngqimkl84p";
+  };
+
+  buildInputs = [
```

Comment:
It seems you forgot to add meson to nativeBuildInputs.

Reply:
Even though meson and ninja are specified on the project page, it also builds with just ./configure and make. I tried to make it work with meson, but for some reason it turned out to be much harder.

Reply:
```
...
Build targets in project: 24
Found ninja-1.8.2 at /nix/store/35wvn5g0igx8xpqzm66j3anpi0icy6rg-ninja-1.8.2/bin/ninja
meson: enabled parallel building
building
build flags: -j1 -l1
[1/66] Generating modem-manager-gui.appdata.xml_merge with a custom command.
[2/66] Generating modem-manager-gui.desktop_merge with a custom command.
[3/66] Generating man-de with a custom command.
FAILED: man/de/modem-manager-gui.1.gz
/build/hg-archive/man/manhelper.py ../man/de/de.po man/de/modem-manager-gui.1.gz
/bin/sh: /build/hg-archive/man/manhelper.py: not found
ninja: build stopped: subcommand failed.
builder for '/nix/store/jnn4w26z9z13kqgij31s0mw0y1p8p2m0-modem-manager-gui.drv' failed with exit code 1
```

Reply:
It cannot find the `/usr/bin/env` in the shebang of the script You will need to do

```
postPatch = ''
  patchShebangs man/manhelper.py
'';
```

possibly preceded by `chmod +x`ing the file.

Reply:
manhelper.py is already executable, so it worked without chmod, but then the script itself failed:

```
meson: enabled parallel building
building
build flags: -j1 -l1
[1/66] Generating modem-manager-gui.appdata.xml_merge with a custom command.
[2/66] Generating modem-manager-gui.desktop_merge with a custom command.
[3/66] Generating man-de with a custom command.
FAILED: man/de/modem-manager-gui.1.gz
/build/hg-archive/man/manhelper.py ../man/de/de.po man/de/modem-manager-gui.1.gz
Traceback (most recent call last):
  File "/build/hg-archive/man/manhelper.py", line 20, in <module>
    for line in intfile:
  File "/nix/store/nrl0l79a48924xb0897ap572xf29ciir-python3-3.6.6/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 363: ordinal not in range(128)
ninja: build stopped: subcommand failed.
builder for '/nix/store/v9v1sbnpr6xazzp8qrr98yb32wnpmyi9-modem-manager-gui.drv' failed with exit code 1
```

Reply:
It is one of issues with our python. You can fix it by adding `LC_ALL = "en_US.utf-8"` attribute to the derivation and `glibcLocales` to `nativeBuildInputs`.

Reply:
That worked. Updated the package to build with meson and ninja.